### PR TITLE
add readiness probe to proxy test

### LINKED ISSUE #21484: e2e flake: Proxy version v1 [It] should proxy through a service and a pod
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/21251/kubernetes-pull-build-test-e2e-gce/29155/build-log.txt

```
Failure [132.205 seconds]
Proxy
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/proxy.go:40
  version v1
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/proxy.go:39
    should proxy through a service and a pod [Conformance] [It]
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/proxy.go:224

    0: path /api/v1/namespaces/e2e-tests-proxy-ehd91/pods/proxy-service-yckdo-cd23f/proxy/ gave error: an error on the server has prevented the request from succeeding

    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/proxy.go:222
```

cc @kubernetes/goog-cluster

### PR DESCRIPTION
**Pull request by lavalamp**
Fixes #21484

### CODE REVIEW

File: test/e2e/util.go [hunk 1]
```diff
@@ -1613,10 +1614,11 @@ func (config *RCConfig) create() error {
...
 						{
-							Name:    config.Name,
-							Image:   config.Image,
-							Command: config.Command,
-							Ports:   []api.ContainerPort{{ContainerPort: 80}},
+							Name:           config.Name,
+							Image:          config.Image,
+							Command:        config.Command,
+							Ports:          []api.ContainerPort{{ContainerPort: 80}},
+							ReadinessProbe: config.ReadinessProbe,
```
Comment (bprashanth) [inline test/e2e/util.go]: Don't we actually need to poll till Ready? Assuming if this method didn't know about readiness before it only polled till Running, but I haven't checked.

**Comment by bprashanth** [comment]
The e2e failure is https://github.com/kubernetes/kubernetes/issues/21769, in case you need a rerun without uploading a new commit

**Comment by ihmccreery** [comment]
Handing over to @bprashanth to review; hand back if you'd rather not.

File: test/e2e/util.go (hunk 1 above)
Comment (lavalamp) [inline test/e2e/util.go]: One might think so, but it actually already has separate runningButNotReady and runningAndReady states that it prints out in the logs.

**Comment by lavalamp** [comment]
@k8s-bot test this flake #21769

(thx @bprashanth)

**Comment by lavalamp** [comment]
@k8s-oncall this is a flake fix, please manually merge

**Comment by lavalamp** [comment]
@k8s-oncall this is a flake fix, please manually merge

**Pull request merged**